docs/x509/reference.rst

```diff
@@ -1119,6 +1119,21 @@ X.509 Extensions
 
         The binary value of the identifier.
 
+    .. classmethod:: from_public_key(public_key)
+
+        .. versionadded:: 1.0
+
+        Creates a new SubjectKeyIdentifier instance using the public key
+        provided to generate the appropriate digest. This should be the public
+        key that is in the certificate.
```

Comment:
Thanks

On Sat, Aug 8, 2015 at 10:49 AM, Paul Kehrer <notifications@github.com>
wrote:

> In docs/x509/reference.rst
> <https://github.com/pyca/cryptography/pull/2204#discussion_r36578919>:
>
> > @@ -1119,6 +1119,21 @@ X.509 Extensions
> >
> >          The binary value of the identifier.
> >
> > +    .. classmethod:: from_public_key(public_key)
> > +
> > +        .. versionadded:: 1.0
> > +
> > +        Creates a new SubjectKeyIdentifier instance using the public key
> > +        provided to generate the appropriate digest. This should be the public
> > +        key that is in the certificate.
>
> That is correct, although it's the only option I've ever seen used. I'll
> add some more prose to explain that it's the SHA1 of the subjectPublicKey
> BIT STRING and link the RFC section
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/2204/files#r36578919>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084